summingbird-batch/src/main/scala/com/twitter/summingbird/batch/Batcher.scala

```diff
@@ -169,7 +169,13 @@ trait Batcher extends Serializable {
     dateToBatch(interval)(truncateUp)(truncateDown)
 
   def toInterval(b: BatchID): Interval[Timestamp] =
-    Intersection(InclusiveLower(earliestTimeOf(b)), ExclusiveUpper(earliestTimeOf(b.next)))
+    if (b.id < Long.MaxValue) {
```

Comment:
I'm not exactly sure what you are proposing here. We could write this code with a match:
```scala
batchIdSuccessible.next(b) match {
  case Some(bnext) => // can do left closed right open
  case None => // can't call next, so can only do inclusive
}
```
I'm not sure what `.map` has to do with it. I think you'd need a `.flatMap` actually since both `earliestTimeOf` might return Option and `next` should return `Option`.

Rather than do a big refactor of batcher and BatchID to deal with the partiality now, I just want to tackle fixing the things we can without changing the binary signature.

Note, the range bug I actually saw in a test (it would not terminate since previously it was an infinite loop).